### PR TITLE
Adding base functionality for the initial load of users

### PR DESCRIPTION
**Pull request by Pokom**
I have added code in the Clique object and the Friends object to handle getting all of the users in the click when the user initially joins.

### CODE REVIEW

File: Cliques.js [hunk 1]
```diff
@@ -38,14 +38,33 @@ Clique.prototype.add = function(ws) {
...
 
 Clique.prototype.onMessage = function(sender, message) {
 	var payload = {}
 	try {
 		payload = JSON.parse(message)
-		payload.id = sender.id
-		payload.name = sender.name
+		switch (payload.event) {
+			case 'friend-joined':
+				//Chris, how can I clean this up? This looks
```
Comment (ChrisMondok) [inline Cliques.js]: That's easy! There's no need to check friends.length. `[].map(prettyMuchAnything)` returns [].

**Comment by ChrisMondok** [comment]
I'm going to tidy a few things up first, but I'm checking it out now.

File: client/Comms.js [hunk 2]
```diff
@@ -3,7 +3,8 @@ function Comms() {
 	this.outgoing = []
 	this.incoming = []
 	this.listeners = []
-
+	this.lastConnection = []
```
Comment (ChrisMondok) [inline client/Comms.js]: Is this used for anything?

File: client/init.js [hunk 3]
```diff
@@ -18,25 +22,46 @@ addEventListener('load', function() {
 
 		form.querySelector('button').disabled = false
 	}
-
+	/**
+	 *
+	 * @param  {string} name The name of the new user, the input. Should probably
+	 *                       validate this.
+	 * @return {stuff}      stuff, not quite sure what stuff is yet.
```
Comment (ChrisMondok) [inline client/init.js:29]: Here, stuff is an array of two resolved promises. One is the comms object, the other is a map. `stuff` is definitely a bad name for this variable, I'm going to change it, but I can't change that this needs to return two things.

File: client/init.js [hunk 4]
```diff
@@ -18,25 +22,46 @@ addEventListener('load', function() {
...
 	}
-
+	function getCurrentUserList(){
+		try {
+			comms.send({
+				event: 'friend-joined',
+				user: name
+			})
+		} catch(e) {
+			console.log(`error getting the users list ${e}`)
```
Comment (ChrisMondok) [inline client/init.js]: Unfortunately, we can't use template literals in client side code yet.

File: client/Comms.js (hunk 2 above)
Comment (Pokom) [inline client/Comms.js]: It is not. I took it out on my local version, just didn't get around to committing.

File: client/init.js (hunk 3 above)
Comment (Pokom) [inline client/init.js:29]: I'm fine with leaving stuff, after I traced it and thought about it, Stuff actually fits. It's just stuff.

File: client/init.js (hunk 4 above)
Comment (Pokom) [inline client/init.js]: Ah damn I forgot about that. Let me take it out.

File: client/FriendsList.js [hunk 5]
```diff
@@ -7,10 +7,23 @@ function FriendsList(map) {
...
+				//don't have to always iterate over the friends list
+				if(!this.friends.find({id: friend.id}))
+					this.addFriend(new Friend(this.map, friend))
+			}, this)
+			break
 		case 'location-updated':
 			if(friend) friend.update(message)
-			else this.addFriend(new Friend(this.map, message))
-			break;
+			break
```
Comment (ChrisMondok) [inline client/FriendsList.js:26]: I'm going to put adding friends back to handle messages being received out-of-order.

**Comment by Pokom** [comment]
I've just pushed a commit that addresses all of the comments.

**Pull request merged**